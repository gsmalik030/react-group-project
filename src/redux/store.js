import { createStore, combineReducers,applyMiddleware } from "@reduxjs/toolkit";
import thunk from "redux-thunk";
import logger from "redux-logger";
import rockets from "./rockets/rockets";
import missions from "./missions/missions";
const rootReducer = combineReducers(
    {
        rockets,
        missions
    }
)
const store = createStore(rootReducer,applyMiddleware(thunk,logger))

export default store;