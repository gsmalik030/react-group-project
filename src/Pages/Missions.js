import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchMissions, toggleMissionReservation } from '../redux/missionsReducer';

const Missions = () => {
  const missionDetails = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!missionDetails.length) dispatch(fetchMissions());
  }, [dispatch, missionDetails]);

  return (
    <main>
      <table className="missions-table">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {missionDetails.map((mission) => (
            <tr key={mission.mission_id}>
              <td className="mission-name">{mission.mission_name}</td>
              <td className="mission-description">{mission.description}</td>
              <td className="member-status-cell"><span className={mission.reserved ? 'active-badge' : ''}>{mission.reserved ? 'Active Member' : 'NOT A MEMBER'}</span></td>
              <td className="status-btn-cell">
                <button
                  type="button"
                  className={mission.reserved ? 'danger-btn' : ''}
                  onClick={() => dispatch(toggleMissionReservation(mission.mission_id))}
                >
                  {mission.reserved ? 'Leave Mission' : 'Join Mission'}
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </main>
  );
};

export default Missions;
