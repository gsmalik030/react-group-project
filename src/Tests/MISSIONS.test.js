import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../Pages/MISSIONS';
import store from '../redux/configureStore';

it('renders correctly', () => {
  const Mission = renderer
    .create(
      <Provider store={store}><Missions /></Provider>,
    )
    .toJSON();
  expect(Mission).toMatchSnapshot();
});
