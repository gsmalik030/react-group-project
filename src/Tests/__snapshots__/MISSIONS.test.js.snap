// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <table
    className="missions-table"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</main>
`;
